src/nvim/terminal.c

```diff
@@ -217,6 +224,8 @@ Terminal *terminal_open(buf_T *buf, TerminalOptions opts)
   vterm_set_utf8(rv->vt, 1);
   // Setup state
   VTermState *state = vterm_obtain_state(rv->vt);
+  // handle custom escape sequences
+  vterm_state_set_unrecognised_fallbacks(state, &vterm_parser_callbacks, rv);
```

Comment:
wonder what the difference is vs `vterm_screen_set_unrecognised_fallbacks` which is used in https://github.com/neovim/neovim/pull/22159 .
looks like they are equivalent:
https://github.com/neovim/libvterm/blob/9d6d2112335080312ef8c36667fa717ded4f7daf/src/screen.c#L1064-L1067